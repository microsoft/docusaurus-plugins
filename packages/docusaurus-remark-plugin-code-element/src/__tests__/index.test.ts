import { describe, it, expect } from 'vitest';
import path from 'path';
import vfile from 'to-vfile';
import mdx from 'remark-mdx';
import remark from 'remark';
import plugin from '../index';
import type { PluginOptions } from '../types';

const processFixture = async (name: string, options: PluginOptions) => {
  const filePath = path.join(__dirname, '__fixtures__', `${name}.md`);
  const file = await vfile.read(filePath);
  const result = await remark().use(mdx).use(plugin, options).process(file);

  return result.toString();
};

const options: PluginOptions = {
  langs: [
    {
      lang: 'dot',
      element: 'Dot',
    },
  ],
};

describe('compile-code plugin', () => {
  it('works on compile file', async () => {
    const result = await processFixture('compile', options);
    expect(result).toMatchSnapshot();
  });
});
