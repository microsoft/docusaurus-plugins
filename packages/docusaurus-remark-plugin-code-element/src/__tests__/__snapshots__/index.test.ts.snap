// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`compile-code plugin > works on compile file 1`] = `
"import Dot from '@theme/Dot';

Some snippets

<Dot lang={\\"dot\\"} meta={{}} value={\\"foo\\"} />

<Dot lang={\\"dot\\"} meta={{\\"meta\\":true,\\"meta1\\":0,\\"meta2\\":\\"hello\\",\\"metaf\\":false,\\"metat\\":true}} value={\\"foo\\"} />
"
`;
